### PR TITLE
ref: Create separate kernel and module images

### LINKED ISSUE #31: Stop building modules into base kernel images
Instead we are going to have images per module "group" which can be mounted in automatically by flintlock.

Requires corresponding change in Flintlock https://github.com/weaveworks-liquidmetal/flintlock/issues/537

### PR DESCRIPTION
**Pull request by Callisto13**
Closes #31 , part of #49 

This commit moves the kernel image out of the `flintlock/` subdir and splits the kernel image into two: one containing just the binary, the other just the modules.

The sequence works like so:
- `builder` creates a starting point image from Ubuntu, installing packages etc.
- `base` uses builder, pulls and compiles the kernel, then saves the modules and the binary in a `scratch` image. I have kept this image under the old `flintlock-kernel` name with modules and the binary to maintain backwards compatibility.
- `modules` and `bin` use `base` and simply copy the things they need into a `scratch` image.

The images produced are:
- `kernel-builder`
- `flintlock-kernel` (kept the same for backwards compatibility)
- `kernel-bin`
- `kernel-modules`

### CODE REVIEW

**Comment by Callisto13** [comment]
Validated cluster works with:

```yaml
rootVolume:
  id: root
  image: "docker.io/claudiaberesford/capmvm-k8s-os:1.23.5"
kernel:
  filename: "boot/vmlinux"
  image: "docker.io/claudiaberesford/kernel-bin:5.10.77"
volumes:
- id: modules
  image: "docker.io/claudiaberesford/kernel-modules:5.10.77"
  mountPoint: /lib/modules/5.10.77
```

**Review by yitsushi — APPROVED** [review]

File: .github/workflows/flintlock-ubuntu-perftest.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  pull_request:
-    paths:
-      - 'flintlock/perf-test/**'
-    branches: [main]
-  push:
-    paths:
-      - 'flintlock/perf-test/**'
-    branches: [main]
+on: {}
+  # pull_request:
```
Comment (richardcase) [inline .github/workflows/flintlock-ubuntu-perftest.yml:4]: nit: can we get rid of these comments?

**Review by richardcase — COMMENTED** [review]
How would we slot the CloudHypervisor kernel into this re-organised structure? It requires different options to be enabled and so it would require a different kernel/modules images

**Comment by Callisto13** [comment]
> How would we slot the CloudHypervisor kernel into this re-organised structure? It requires different options to be enabled and so it would require a different kernel/modules images

I planned to punt that to my stretch goal of using something like Packer to build our images instead. From what I see it provides an easier way of reusing modules of build parts into a matrix so we don't have to dupe things. If not that, I can split what we have further into as many reusable parts as possible and have like `kernel/firecracker` and `kernel/cloudhypervisor` subdirs:

```bash
# all off the top of my head
- kernel
  - builder
  - common base elements
  - firecracker
    - configs 
    - modules
    - bin
  - cloud hypervisor
    - configs
    - modules
    - bin
```

File: .github/workflows/flintlock-ubuntu-perftest.yml (hunk 1 above)
Comment (Callisto13) [reply .github/workflows/flintlock-ubuntu-perftest.yml:4]: these test the base-ubuntu image, which is being removed in #52, so i will add removal of the perf image too

**Comment by Callisto13** [comment]
@richardcase do you want me to factor in experimental images as part of this pr or can i leave it for the next one?

**Comment by richardcase** [comment]
> @richardcase do you want me to factor in experimental images as part of this pr or can i leave it for the next one?

Lets leave it as a follow-up PR.

**Review by richardcase — APPROVED** [review]

**Review by yitsushi — APPROVED** [review]

**Pull request merged**